akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -100,8 +151,17 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
         "`ORMap.put` must not be used to replace an existing `ORSet` " +
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
-    else
-      new ORMap(keys.add(node, key), values.updated(key, value))
+    else {
+      val newDelta = delta match {
+        case Some(d) ⇒
+          d.merge(PutDeltaOp(keys.resetDelta.add(node, key).delta.get, Map(key → value), node))
+        case None ⇒
+          PutDeltaOp(keys.resetDelta.add(node, key).delta.get, Map(key → value), node)
```

Comment:
`PutDeltaOp(keys.resetDelta.add(node, key).delta.get, Map(key → value), node)` is the same for both cases so can be defined in a `val` before the `match`